subsys/spm/secure_services.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <cortex_m/tz.h>
+
+#if defined(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
+
+extern void sys_arch_reboot(int type);
+
+static void spm_system_reboot(void)
+{
+	sys_arch_reboot(0);
+}
+
+/** @brief Secure Service for issuing a System Reset
+ *
+ *
+ * Secure Entry function for allowing Non-Secure firmware
+ * to request a System Reset. The function simply calls the
```

Comment:
I think maybe the comment "Simply calls Zephyr System Resets" probably should state "calls the spm_system_reboot functions". Issuing a reset will probably need a "reset manager" and synchronization of both Non-Secure/Secure code regions to prevent loss of data. If we add another symbol in-between it should be fairly straightforward to replace the default reboot strategy in special cases

Reply:
Moved some docs around and generalized the wording.